[BUG] org.opensearch.index.shard.SegmentReplicationIndexShardTests.testReplicaReceivesGenIncrease flaky

```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.index.shard.SegmentReplicationIndexShardTests.testReplicaReceivesGenIncrease" -Dtests.seed=AF518467E7BD5FFF -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=en-MT -Dtests.timezone=Pacific/Wallis -Druntime.java=19

org.opensearch.index.shard.SegmentReplicationIndexShardTests > testReplicaReceivesGenIncrease FAILED
    java.lang.AssertionError: expected:<0> but was:<31>
        at __randomizedtesting.SeedInfo.seed([AF518467E7BD5FFF:223BC64F10A3EBDF]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.opensearch.index.shard.SegmentReplicationIndexShardTests.testReplicaReceivesGenIncrease(SegmentReplicationIndexShardTests.java:276)
```

https://github.com/opensearch-project/OpenSearch/pull/5223

--- a/test/framework/src/main/java/org/opensearch/index/shard/IndexShardTestCase.java
+++ b/test/framework/src/main/java/org/opensearch/index/shard/IndexShardTestCase.java
@@ -1301,7 +1301,8 @@ public abstract class IndexShardTestCase extends OpenSearchTestCase {
                 }
             );
             ids.add(target);
-            countDownLatch.await(1, TimeUnit.SECONDS);
+            countDownLatch.await(30, TimeUnit.SECONDS);
+            assertEquals("Replication should complete successfully", 0, countDownLatch.getCount());
         }
         return ids;
     }

--- a/test/framework/src/main/java/org/opensearch/index/shard/IndexShardTestCase.java
+++ b/test/framework/src/main/java/org/opensearch/index/shard/IndexShardTestCase.java
@@ -1301,9 +1301,9 @@ public abstract class IndexShardTestCase extends OpenSearchTestCase {
                 }
             );
             ids.add(target);
-            countDownLatch.await(30, TimeUnit.SECONDS);
-            assertEquals("Replication should complete successfully", 0, countDownLatch.getCount());
         }
+        countDownLatch.await(30, TimeUnit.SECONDS);
+        assertEquals("Replication should complete successfully", 0, countDownLatch.getCount());
         return ids;
     }
 

